src/main/java/com/blackducksoftware/integration/hub/rest/CredentialsRestConnection.java

```diff
@@ -0,0 +1,39 @@
+package com.blackducksoftware.integration.hub.rest;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+
+import org.apache.commons.lang3.StringUtils;
+import org.restlet.Context;
+import org.restlet.resource.ClientResource;
+
+import com.blackducksoftware.integration.exception.EncryptionException;
+import com.blackducksoftware.integration.hub.exception.BDRestException;
+import com.blackducksoftware.integration.hub.global.HubProxyInfo;
+import com.blackducksoftware.integration.hub.global.HubServerConfig;
+
+public class CredentialsRestConnection extends RestConnection {
+
+	public CredentialsRestConnection(final HubServerConfig hubServerConfig)
+			throws IllegalArgumentException, URISyntaxException, BDRestException, EncryptionException {
+		super();
+		setBaseUrl(hubServerConfig.getHubUrl().toString());
+		final HubProxyInfo proxyInfo = hubServerConfig.getProxyInfo();
+		if (proxyInfo.shouldUseProxyForUrl(hubServerConfig.getHubUrl())) {
+			setProxyProperties(proxyInfo);
+		}
+		setTimeout(hubServerConfig.getTimeout());
+		final String userName = hubServerConfig.getGlobalCredentials().getUsername();
+		final String password = hubServerConfig.getGlobalCredentials().getEncryptedPassword();
+		if (StringUtils.isNotBlank(userName) && StringUtils.isNotBlank(password)) {
```

Comment:
We'd want to use the encrypted password to test for existence (it's the actual thing in the object) and use the decrypted password for connecting (since that can throw EncryptionException)